File: backend/src/main/java/com/example/backend/service/AdminService.java
package com.example.backend.service;

public interface AdminService {
    //quản lí người dùng
    org.springframework.data.domain.Page<com.example.backend.model.respon.UserResponse> getAllUsers(org.springframework.data.domain.Pageable pageable);
    org.springframework.data.domain.Page<com.example.backend.model.respon.UserResponse> filterUsersByActive(Boolean isActive, org.springframework.data.domain.Pageable pageable);
    org.springframework.data.domain.Page<com.example.backend.model.respon.UserResponse> filterUsersByRole(com.example.backend.model.enums.Role role, org.springframework.data.domain.Pageable pageable);
    Boolean lockOrUnlockUser(Long userId, Boolean isActive);

    //quản lí bài viết
    org.springframework.data.domain.Page<com.example.backend.model.respon.PostResponse> getAllPosts(org.springframework.data.domain.Pageable pageable);
    org.springframework.data.domain.Page<com.example.backend.model.respon.PostResponse> filterPostsByStatus(com.example.backend.model.enums.PostStatus status, org.springframework.data.domain.Pageable pageable);
    Boolean deletePost(Long postId);

    //báo cáo thống kê
    Long countNewUsers(java.time.LocalDateTime startDate, java.time.LocalDateTime endDate);
    Long countActiveUsers();
    Long countNewPosts(java.time.LocalDateTime startDate, java.time.LocalDateTime endDate);
    Long countNewComments(java.time.LocalDateTime startDate, java.time.LocalDateTime endDate);
    Long countNewReacts(java.time.LocalDateTime startDate, java.time.LocalDateTime endDate);
}

File: frontend/src/service/impl/AdminService.ts
import { http } from "../http/http";
import type { Role, PostStatus } from "../../utils/EntityType";
import type { ApiDataResponse, Page, UserResponse, PostResponse } from "../../utils/ResponType";

export const getAllUsers = (page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<UserResponse>>>(`/admin/users?page=${page}&size=${size}`);
};

export const filterUsersByActive = (isActive: boolean, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<UserResponse>>>(
        `/admin/users/filter/active?isActive=${isActive}&page=${page}&size=${size}`
    );
};

export const filterUsersByRole = (role: Role, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<UserResponse>>>(
        `/admin/users/filter/role?role=${encodeURIComponent(role)}&page=${page}&size=${size}`
    );
};

export const lockOrUnlockUser = (userId: number, isActive: boolean) => {
    return http.put<ApiDataResponse<boolean>>(`/admin/users/${userId}/status?isActive=${isActive}`);
};

export const getAllPosts = (page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<PostResponse>>>(`/admin/posts?page=${page}&size=${size}`);
};

export const filterPostsByStatus = (status: PostStatus, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<PostResponse>>>(
        `/admin/posts/filter/status?status=${encodeURIComponent(status)}&page=${page}&size=${size}`
    );
};

export const deletePost = (postId: number) => {
    return http.delete<ApiDataResponse<boolean>>(`/admin/posts/${postId}`);
};

export const getStatistics = (startDate: string, endDate: string) => {
    return http.get<ApiDataResponse<Record<string, number>>>(
        `/admin/statistics?startDate=${encodeURIComponent(startDate)}&endDate=${encodeURIComponent(endDate)}`
    );
};
